websites/documentation/src/content/writing/self-learning.mdx

```diff
@@ -137,3 +139,351 @@ Examples
+- `isLoading`: It's a boolean value indicating if the fetch operation is still in progress (if `true`)
+- `error`: In case of error, this property will contain error information. The format of error might depend on its type.
+- `data`: If the fetch operation completes successfully, this property will contain an array of all the courses in which the user is erolled in with the status
+
+The hook can be used in conjunction with the `getCourseStatusByCourseId` helper to retrieve a specific course status
+
+Example:
+
+```javascript
+const { data, isLoading } = useFetchCourses();
+
+useEffect(() => {
+   if (!isLoading && data) {
+     getCourseStatusByCourseId(data?.result?.enrolledCourses, props.courseId)
+   }
+  }, [data, isLoading, error]);
+```
+
+**Note:** This hook uses [SWR](https://swr.vercel.app/) library to interact with the underlying API. This means it gets all the automatic features that comes with SWR.
+
+
+### SidebarCourseStatus
+
+This component purpose is displaying an icon next to the course title in the navigation bar. The icon displayed depends on the course status and will only be displayed if the user is logged in.
+
+The component expects the following props:
+- `courseId`: a number representing the id of the Moodle course
+
+Example:
+```javascript
+
+<LinkItemWithIcon>
+  <SidebarCourseStatus courseId={516} />
+  <LinkTitle>The course title</LinkTitle>
+</LinkItemWithIcon>
+```
+
+### SidebarTopicStatus
+
+This component purpose is displaying an icon next to the course's topics titles in the navigation bar. The icon displayed depends on the topic status and will only be displayed if the user is logged in.
+
+The component expects the following props:
+- `courseId`: a number representing the id of the Moodle course
+- `pageTitle`: the title of the topic (used to find the topic status within the CourseDetails object)
+
+Example:
+```javascript
+
+<SidebarTopicStatus
+  courseId={courseId}
+  pageTitle={currTopicName}
+/>
+```
+
+### UserProfileInit
+
+This is a component that doesn't render any html, in a standard SPA it would be a hook.
+Its purpose is to monitor the user profile, either coming from Auth0 at the initial login, or stored in `LearningContext`, and check if at any point the profile becomes incomplete,
+meaning that some defined fields are missing. When this situation happens, the complete profile modal is triggered.
+Another function of this component is to keep in sync the user profile with the version of user profile stored into the LearningContext.
+
+The component doesn't recieves any props and it's just injected at the root of the application.
+
+### ProfileModal
+
+When triggered, this component renders a modal with a form containing some user profile fields. The user is able to modify their values and
+submit the form once the changes are completed.
+The profile modal visibility is bound to a value in the LearningContext and the correct way to open it or close it is by useCourseInfoByPageSlugs
```

Comment:
I'm not sure if I understand this sentence. `useCourseInfoByPageSlugs` is not really fitting into the sentence, or maybe I just don't get it 😅

Reply:
indeed, it's probably a copy/paste glitch. Thank you